### ISSUE TITLE
'PkgSet' object has no attribute 'namedict'

### ISSUE THREAD

**Issue by nilxam** [openSUSE/product-composer#11]
```
[   40s] Traceback (most recent call last):
[   40s]   File "/usr/bin/product-composer", line 8, in <module>
[   40s]     sys.exit(main())
[   40s]              ^^^^^^
[   40s]   File "/usr/lib/python3.11/site-packages/productcomposer/cli.py", line 86, in main
[   40s]     args.func(args)
[   40s]   File "/usr/lib/python3.11/site-packages/productcomposer/cli.py", line 141, in build
[   40s]     create_tree(args.out, product_base_dir, yml, pool, flavor, args.vcs, args.disturl)
[   40s]   File "/usr/lib/python3.11/site-packages/productcomposer/cli.py", line 259, in create_tree
[   40s]     link_rpms_to_tree(maindir, yml, pool, arch, flavor, debugdir, sourcedir)
[   40s]   File "/usr/lib/python3.11/site-packages/productcomposer/cli.py", line 876, in link_rpms_to_tree
[   40s]     main_pkgset = create_package_set(yml, arch, flavor, 'main', pool=pool)
[   40s]                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[   40s]   File "/usr/lib/python3.11/site-packages/productcomposer/cli.py", line 855, in create_package_set
[   40s]     pkgset.sub(pkgsets[oname])
[   40s]   File "/usr/lib/python3.11/site-packages/productcomposer/core/PkgSet.py", line 42, in sub
[   40s]     otherbyname = other.namedict()
[   40s]                   ^^^^^^^^^^^^^^
[   40s] AttributeError: 'PkgSet' object has no attribute 'namedict'
```

I suppose it tries to do a reordering but I can not find `namedict()` being defined anywhere

**Comment by nilxam** [comment]
@adrianschroeter can you please have a look?

**Comment by adrianschroeter** [comment]
fixed by Michael. Submitted together another fix as 0.4.16
